### PR TITLE
Adding content-type header to POST requests to avoid OPTIONS requests

### PR DESCRIPTION
**Pull request by indraastra**
This fix addresses the last remaining (client-side) issue after @m-mcgowan's work in https://github.com/spark/firmware/pull/694.

I've tested this out from the node cli on Yosemite and also when packaged up within [softap-setup-meteor](https://github.com/msolters/softap-setup-meteor) on Android and iOS both, and it seems to work in all cases.

### CODE REVIEW

**Comment by msolters** [comment]
:+1: glad to hear it works!

**Comment by affinity-matrix** [comment]
beautiful! :heart:

**Comment by affinity-matrix** [comment]
@indraastra, @msolters: would you both please add yourselves as contributors in the `package.json` file?

Thanks for your help!

**Comment by msolters** [comment]
Sure.  I wonder if it would be prudent at this stage to add in the `withCredentials: false`?

I haven't been able to test @indraastra's changes directly on the machine I had that was throwing those errors.  I do know that it can help -- but I'm not sure if it will cause any conflicts with the `Content-Type` being used here.

**Comment by affinity-matrix** [comment]
@msolters I'm happy to make that a thing!

Do you think it would work if we made it an option like `noCredentials`?

**Comment by msolters** [comment]
I'm not sure what you mean, exactly.  What is `noCredentials`?  And what do you mean by use it as an option?

`withCredentials` is a parameter that translates into a header that we are specifying to ensure no rogue clients decide to tack on credentials flag to the HTTP requests when they are sent over to the Photon.  I don't see it as an option really, more of an attempt to guarantee proper request construction.

Bottom line -- we should test it, and then move from there.

**Comment by affinity-matrix** [comment]
@msolters right, that sounds like a plan then!

**Comment by msolters** [comment]
I've just tested 3.0.2 on the same machine that had thrown those credential-match errors, and I'm not getting them anymore.  I'd say that, for now, we're all clear to move on without forcing the credential flag.

**Comment by affinity-matrix** [comment]
Fantastic! Thank you for your help!
